Fix Validate button for DB backup

himdel opened the pull request:
for details, please see the comments (especially brandon's) in PR #1276, but `log_depot_edit` is used only for *log depot* validation, the rest should be still using `log_depot_validate` (even though the name *might* be considered a wee bit misleading).

Also removing executable bit from _x_gtl.

EDIT: fixes https://bugzilla.redhat.com/show_bug.cgi?id=1218406

blomquisg commented:
Stalled travis build: https://github.com/ManageIQ/manageiq/issues/2333

dclarizio commented:
@himdel I'm not sure how to verify this fix.  Can you walk @martinpovolny through it so he can verify/merge?  Thx, Dan

himdel commented:
@martinpovolny: added fixes to the actual Verify functionality, should work fine now (except smb always fails with unmount error because it has no rights to create the mountpoint, but that's a separate issue),

martinpovolny commented:
@himdel : travis seems to dislike your changes, please, check

dclarizio commented:
@himdel if you get a chance, can you address the test failures?

himdel commented:
I think it's the test that's wrong with this one, `validate_settings` seems to *throw* on error, both in ftp (from connect) and smb (explicitly). It never actually returns a bool either.. My changes to `verify_depot_hash` reflect that behaviour and no longer failed on actual false.

I'd rather see `validate_settings` return true or false but we actually depend on the exception message being available... So I changed `verify_depot_hash!` to throw on false, and added a test ensuring we can handle both false and an exception.

AparnaKarve commented:
@himdel @dclarizio I added 2 commits that address the Validate button crash issue in Schedule Editor and in diagnostics database tab.
The crash is addressed, however as @himdel mentioned earlier, the validation always fails for smb. 

Here is a screenshot of the smb mount error that I'm seeing (looks like its complaining about the mount command usage ) -
![screen shot 2015-04-24 at 11 14 10 am](https://cloud.githubusercontent.com/assets/1538216/7326994/87cbeb3a-ea7e-11e4-94ee-971f30f523e2.png)

dclarizio commented:
@martinpovolny Can you please verify and merge if this looks good.  Chatted with Brandon regarding the smb error, so we should be able to let that be as long as the UI issues are handled:
```[1:08:42 PM] Brandon Dunne: i don't think you'll be able to mount smb or nfs unless you're on an appliance with the way the library is currently written```

martinpovolny commented:
@himdel : can you please, rebase? I'd do final nesting and merge this

himdel commented:
@martinpovolny ugh, there's a collision with #2657 (well, a618001) - I'll have to redo most of it. It broke my hack that made it not care whether FileDepot*.validate_settings return false or throw on failure :). (Because FileDepotSmb always returns false, while FileDepotFtp always throws.) (And the 2 places in 4ba556e both expect throws.) => I'll try to finish it tomorrow.

martinpovolny commented:
@himdel : no prob, bugz have priority; sorry for the delay

himdel commented:
Looked a bit further, seems no there are no other callers of the FileDepot* validation except for  `app/controllers/ops_controller/diagnostics.rb` calling `validate_settings` and `MiqSchedule.verify_depot_hash` calling `verify_credentials`. Both cases assume an exception on error, FileDepotSmb already always throws... made FileDepotFtp#verify_credentials throw as well. Should be ready and mergeable.

dclarizio commented:
@martinpovolny I think this is ready to be verified and merged if ok.  Thx, Dan
